src/node/node.h

```diff
@@ -13,11 +13,25 @@ class Chain;
 class ChainClient;
 } // namespace interfaces
 
-//! Pointers to interfaces used during init and destroyed on shutdown.
+//! Node struct containing references to chain state and connection state.
+//!
+//! This is used by init, rpc, and test code to pass object references around
+//! without needing to declare the same variables and parameters repeatedly, or
+//! to use globals. More variables could be added to this struct (particularly
+//! references to validation and mempool objects) to eliminate use of globals
+//! and make code more modular and testable. The struct isn't intended to have
+//! any member functions. It should just be a collection of references that can
+//! be used without pulling in unwanted dependencies or functionality.
 struct Node
```

Comment:
nit: maybe it should be name NodeState to avoid confusion with `Node` the already-there interface class?

Reply:
>  I guess I have some aversion to adding suffixes like Info Data State to type names because you could add those suffixes to pretty much any type name just making it more verbose without changing the meaning.

Ah, you're right but most of the times it's the cheapest way to avoid name collisions!

+1 for `NodeInterface` if we have opportunity later